MapboxCoreNavigation/LegacyRouteController.swift

```diff
@@ -280,7 +284,10 @@ open class LegacyRouteController: NSObject, Router, InternalRouter, CLLocationMa
         let step = stepProgress.step
         
         //Increment the progress model
-        let polyline = Polyline(step.coordinates!)
+        guard let polyline = step.shape else {
+            return
```

Comment:
Add a precondition failure here, because the lack of a shape is a serious internal consistency error. `RouteStep.shape` is now mutable unlike before.